### DISCUSSION TITLE
user registration with react query

### DISCUSSION THREAD

**Discussion by AbyerAli** [TanStack/query#5106] [General] [answered]
I am new to query environment so need some help
gimme a general idea of how user login and signup is implemented in react query

right now i am just loging in user via useMutation 
code:

```
export const useGetBuyerData = () => {
  const loginBuyer = async ({ email, password }) => {
    const response = await axios.post("/api/users/userLogin", { email, password });
    return response.data;
  }
  return useMutation(loginBuyer);
}

const {
    mutate,
    data: loginData,
 } = useRegisterBuyer();

```
how is a centralized state achived here so my user data is saved and how the registration is impleted ?

**Comment by TkDodo** [comment, marked as answer]
same as any other mutation/query combination: invalidate the query that fetches the information of the logged-in user after login.
